soh/src/code/audio_load.c

```diff
@@ -1499,6 +1522,13 @@ s32 AudioLoad_SlowLoadSeq(s32 seqId, u8* ramAddr, s8* isDone) {
     size_t size;
 
     seqId = AudioLoad_GetRealTableIndex(SEQUENCE_TABLE, seqId);
+    u16 newSeqId = SfxEditor_GetReplacementSeq(seqId);
+    if (seqId != newSeqId) {
+        gAudioContext.seqToPlay[SEQ_PLAYER_BGM_MAIN] = newSeqId;
+        gAudioContext.seqReplaced[SEQ_PLAYER_BGM_MAIN] = 1;
+        Audio_QueueSeqCmd(0x00000000 | ((u8)SEQ_PLAYER_BGM_MAIN << 24) | ((u8)(0) << 0x10) | (u16)seqId);
```

Comment:
looks like `Audio_QueueSeqCmd` takes in a `u32`, so i'm not sure why `0x00000000` is being used for 32 bits

also, it'd be nice to have a comment explaining this line a little bit, also replacing the `24` with `0x18` for consistency (or replacing the `0x10` with `16`, dealers choice)

 from what i'm reading
the last 16 bits are the seq id
the 8 bits before that are always 0 (not sure why)
the 8 bits before that are `SEQ_PLAYER_BGM_MAIN` (not sure why)

Reply:
Each digit of a hex number is 4 bits, and there's 8 digits there so 32 bits, although it probably doesn't need to be there tbh, may have just been to make it clear that the most significant 8 bits (which get parsed from the command as the operation to perform) are 0. When QueueSeqCmd is used in other places for other operations, it follows a similar format to this.

Technically only the last 8 bits get parsed as the sequence ID, the first 8 get parsed as a parameter of some kind (exactly what that parameter means varies based on the operation, in this case I think it gets used as a fade timer. Casting the seqId to a u16 ensures that fade timer is 0 (since I believe this always gets the old seqId which is never more than 8 bits in practice).

I think the next 8 bits that are always 0 here are just unused for this operation.

The next 8 bits that are `SEQ_PLAYER_BGM_MAIN` here determine which sequence player to play the sequence (from the last 8 bits) in. There are 4 sequence players, one for MAIN (heard most often) one for SUB (Used for enemy bgm and IIRC also used for lost woods music when near the goron city shortcut and vice-versa), one for Fanfares and one for SFX.

As I said earlier, the remaining 8 bits at the beginning are for the operation, in this case 0, which is "play sequence immediately" according to the comments in `Audio_ProcessSeqCmd`.

IIRC I copied and slightly adapted this from the `Audio_StartSeq` macro which is not available in this file and making it available caused some conflicts I believe. Since this is in `AudioLoad_SlowLoadSeq`, which in the context of sequence shuffling is only relevant for intercepting Hyrule Field Morning, I figured this copy-paste was fine rather than going down the rabbit hole of resolving whatever include shenanigans were going on or copying the whole macro over.

Reply:
> Each digit of a hex number is 4 bits, and there's 8 digits there so 32 bits

not sure why that threw me off when I saw it the first time

> it probably doesn't need to be there tbh

agreed

> I can make a comment explaining that particular call to QueueSeqCmd

that sounds perfect!